### PR TITLE
MRG: Reconfigure website

### LINKED ISSUE #4249: "What's new" isn't easy to find
I tried looking around for the what's new page but couldn't find it. Where is it discoverable with the current website?

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #4249.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
https://4581-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/index.html

**Comment by choldgraf** [comment]
Can we call it "All of the new things that are coded in MNE-python"?

JK

I like the new layout...one quick thought, there's some weird stuff going on when you make the screen super wide:

image

**Comment by agramfort** [comment]
can you insert an INRIA logo:

https://www.inria.fr/var/inria/storage/images/medias/inria/images-corps/logo-inria-sans-signature-couleur/404509-2-fre-FR/logo-inria-sans-signature-couleur.jpg

while you're at it?

**Comment by choldgraf** [comment]
what about BIDS? :)

https://bids.berkeley.edu/sites/all/themes/bids/logo.png

**Comment by Eric89GXL** [comment]
Sure. It looks like it auto-shrinks if the banner becomes too large anyway

**Comment by mmagnuski** [comment]
Works for me well for the normal window size, but the rescaling is weird when the window gets smaller (but it is not a new problem):
mne page pic 01

**Comment by Eric89GXL** [comment]
Logos updated with Inria and BIDS. Let's fix the UI bugs in another PR when someone gets sufficiently motivated.

**Comment by choldgraf** [comment]
that's fine with me!

**Comment by Eric89GXL** [comment]
Ready to go from my end

https://4583-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/index.html

**Comment by agramfort** [comment]
I personally think that to make the what's new page more visible we lost
many direct links previously on the right column (link to mailing list, how
to cite, etc).

I think I prefer the current layout...

**Comment by Eric89GXL** [comment]
I think that the landing page should really be about getting people interested in / introduced to the package, so adding more links actually ends up detracting from the page. I can add them back in if they are seen as essential top-level links, but I'd rather make them more prominent on the Documentation page instead.

**Comment by choldgraf** [comment]
So, I apologize ahead of time because this is potentially creating more work for folks, but I'm gonna throw out a thought:

Why don't we re-work MNE's front page? Hear me out...

The current website of MNE is *great*, but I think we can do a much better job of prioritizing and organizing information on the front page. I propose that we model this off of the scikit-learn front page. We don't need to do everything the same, but the SKL frontpage is really about getting straight to the point: what does this package let you do? They've got visuals on the frontpage which are great for drawing people in, and links to immediately-relevant information for new (or old) users.

E.g., this is what pops up on my screen when I go to the SKL website:

image

* tons of information about what scikit-learn does
* visuals that look nice and give a feeling for what scikit-learn does
* lots of larger-font text that pops out
* much fewer "small" text
* almost all pieces of information have a link to something else on the website.
* funding groups at the bottom

Compare that with the MNE

image

* ~50% just MNE branding + funding information. Neither of which is useful to people.
* The only information that tells us what MNE does is that bullet list of HTML text w/o any links
* No headers etc that let users visually group information more easily
* 20 lines of code that isn't that useful unless you know what it actually produces.

What I'd propose is that we:

1. Replace the "20 lines of code" with an image w/ a caption that says "reproduce this figure with 20 lines of code" that is clickable and takes you to an example.
2. Even better, have a sliding set of figures, each of which can be recreated w/ < 20 lines of code.
5. Make the "topics" list as a grid of topics w/ larger font, each item in the grid is clickable and takes you to the documentation for that topic. Maybe one button per section in the tutorial?
6. Switch the twitter feed to something other than MNE-python, since that's just a bot that parrots github activity. better would be MNE-news.
7. Move all of the logos + funding information to the bottom.

I'm not saying that we should do this in the current PR, but what do people think about this? I can open an issue (or a PR) to look into this if folks are friendly to it.

**Comment by Eric89GXL** [comment]
+100 from me

You can start with this branch as your basis and open a new PR. At least you'll get the updated logos :)

**Comment by choldgraf** [comment]
Looks like this is where sklearn is doing their carousel code:

https://github.com/scikit-learn/scikit-learn/blob/master/doc/themes/scikit-learn/layout.html#L136

Think we could put together a list of images in the examples that would be worth showing off, and that highlight the diversity of MNE functionality?

**Comment by Eric89GXL** [comment]
Yes, we can even start from the list of bullet points on the main page. But we need to see if others are on board first.

**Comment by choldgraf** [comment]
Agreed - I'll hold off on opening a PR until I get a plus 1 from the grand pooh-bah at least ;-)

**Comment by agramfort** [comment]
can you do a mockup with a slide on google doc so we can suggest / comment?

**Comment by choldgraf** [comment]
@agramfort something like this: https://docs.google.com/presentation/d/1LJIh_mL0FhYa-nWBsPVHJ8Pj9NBUXCUTIVf13GkuK_A/edit?usp=sharing

**Comment by Eric89GXL** [comment]
LGTM

**Comment by teonbrooks** [comment]
@choldgraf I like it :)

**Comment by agramfort** [comment]
I like it too. I'd like also to have quick access to what we presently call
"Help" mailing list and how to contacts ie what sklearn calls "Community"

**Comment by Eric89GXL** [comment]
But let's not take up too much space with it :) Maybe we can have a "Community" button that has a drop-down with links to github/gitter/mailing list. @choldgraf sounds like you have a green light!

**Comment by jasmainak** [comment]
It's cool! But I'm concerned that there is no link to "Download" or "Installation" on the home page. It's usually the first thing I look for when I visit a software page ...

**Comment by Eric89GXL** [comment]
@jasmainak there is a "Get started" as the first option in the top title bar, is it not sufficient?

**Comment by jasmainak** [comment]
Maybe :) "Get started" is a little obscure to me. I'd prefer "Installation".

**Comment by Eric89GXL** [comment]
It's a bit more than just installation, but I can live with it. Though I'd prefer "Install" so it's shorter in the top bar but still conveys the same meaning.

**Comment by jasmainak** [comment]
yeah but I don't understand why installation instructions are nested two levels deep. [This page](https://martinos.org/mne/stable/install_mne_python.html) and [this page](https://martinos.org/mne/stable/install_mne_c.html) should really be linked from the home page.

**Comment by Eric89GXL** [comment]
We don't want to risk cluttering the landing page. Let's do Chris's
modification since we've converged on that, and you can follow up with a PR
proposal to change the link structure (since it's mostly an orthogonal
change)

**Comment by choldgraf** [comment]
@Eric89GXL I spent a bit of time today to try and prototype what this'd look like in the codebase. I just pushed these changes to your fork in case you'd like to iterate on this a little bit as well. Does that sound good?

Here's where it stands right now:

http://predictablynoisy.com/mne-python/

As I said, a long way to go but I think it should be easier to iterate from here. I'm also linking to thumbnails from random examples currently because I didn't want to have to build all of the examples.

**Comment by Eric89GXL** [comment]
Okay, made some changes:

https://4633-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/index.html

Notably:

1. Aesthetic changes relative to what @choldgraf had.
2. Fixed the quick-link buttons @choldgraf had put in to properly unfold the correct documentation section (to satisfy @kingjr's desire to link to the doc directly on the homepage).
3. Added `whats_new` and other important quick-links to footer (to satisfy @agramfort).
4. Changed "Getting started" to "Install" (to satisify @jasmainak).

See what you think.

**Comment by kingjr** [comment]
Very nice! I really like this new organization, it's much more functional. 

A couple of minor remarks
* can you add a `v` mark on the main topics to suggest that it's clickable
* I'm not a big fan of the image selection on the front page. Several are pixelated. If possible I would change the frontpage layout such that we don't need to scroll down for a 1024*768 screen.
* What is the meaning of the logos related to Speed, usability etc?

Again, very nice!

**Comment by Eric89GXL** [comment]
> can you add a v mark on the main topics to suggest that it's clickable

I'm not sure what you mean.

> I'm not a big fan of the image selection on the front page. Several are pixelated. If possible I would change the frontpage layout such that we don't need to scroll down for a 1024*768 screen.

I'll try to trim a little bit of space.

> What is the meaning of the logos related to Speed, usability etc?

Just some eye candy, but clearly they didn't translate properly. I'll try to find some more relevant ones.

**Comment by choldgraf** [comment]
Yo! Some thoughts:

+1s
* I really like the text you've put into the examples and the general look of the slider now.
* I like the idea of the little callout things to the right. (though I wonder if we should come up with something other than "reliability" as most users probably don't think about this kind of thing. It feels a bit more like it's catering to devs rather than avg scientists. Why not something like "Flexibility" and the sub-text would describe how it is meant to handle many workflows and types of data.)
* The footer section w/ the institutions is nice! 
* I also like the formatting of the financial support section.

**Comments / suggestions**
* The number of dots in the slider doesn't match the number of slider items.
* The whitespace around MNE looks a little awkward to me. What if we went with a fixed-width layout that kept the look of MNE consistent across screen sizes. Maybe 800-900px? nts / suggestions
* I'd be a fan of making those call-out points to the right be the same height as the MNE logo, and the text below that's more verbose would span the whole width of the above two elements. Or to have those bullet points scale so their height is always the same as the logo + text height. (or as mentioned above, we just go with fixed width so it always looks the same).
* The slider is below-the-fold on my screen. Perhaps we can shrink the whitespace at the top. At the least I want to make sure that text and "try it" button of each slider image is visible on a ~13" screen.
* I'm not sure how I feel about the buttons changing their width based on the length of the text inside. Were you not a fan of fixed-width buttons?

I think that most of the visual components are here...we just need to improve the layout / spacing a bit and make sure that the most important information is prioritized in terms of what the user sees.

**Comment by choldgraf** [comment]
ah, and to @kingjr 's point about the image slider, I think we should select examples we want to have in the slider and make sure that those examples output images that are higher-res than normal.

**Comment by Eric89GXL** [comment]
> I'm not sure how I feel about the buttons changing their width based on the length of the text inside. Were you not a fan of fixed-width buttons?

They made the layout harder to follow IMO. Feel free to play with it locally

**Comment by choldgraf** [comment]
I agree w/ @Eric89GXL about not getting everything perfect. It will be easy to iterate on specific examples etc once the main machinery is there. I think this PR should focus on layout / broad design principles.

But in another thread I agree we should discuss specific examples we want to show (e.g. there should definitely be a `joint` plot displayed or something like this). We should think "for the most common use-cases of analysis in EEG/MEG/ECoG, what is the single coolest-looking viz we can come up with?"

**Comment by Eric89GXL** [comment]
> I'd be a fan of making those call-out points to the right be the same height as the MNE logo, and the text below that's more verbose would span the whole width of the above two elements. Or to have those bullet points scale so their height is always the same as the logo + text height. (or as mentioned above, we just go with fixed width so it always looks the same).

I tried a ton of iterations (this included, I think?) and the current one seemed to work best. Feel free to try something else locally.

I'll see if I can make the images a bit nicer, but if I can't I'd say let's do that in a follow-up PR. (I don't mind the light pixelation/upsampling personally.)

I'll take a stab at the height/whitespace, though, since it would be nice to fit on a single page.

**Comment by choldgraf** [comment]
My 2 cents, this:

image

Is more in-line with modern web design principles :-)

**Comment by Eric89GXL** [comment]
That's fine, I just want them to span the horizontal space. So feel free to push that button mod. Or probably better to make a PR so we don't conflict (I'm going to tackle whitespace now).

**Comment by choldgraf** [comment]
but TBH I think that a lot of my comments about the call-out buttons etc would take a backseat if we could figure out a way to reduce the whitespace around the logo and to make the slider more pronounced above the fold

**Comment by choldgraf** [comment]
> Or probably better to make a PR so we don't conflict (I'm going to tackle whitespace now).

+1 , I'll wait until this gets merged and then tackle some of the smaller visual stuff

**Comment by choldgraf** [comment]
(and apologies if any of this is annoying feedback...I tend to have strong opinions about design :-P)

**Comment by Eric89GXL** [comment]
Okay I squeezed some space, tweaked some names/titles, and made the images less pixelated (by shrinking the carousel):

https://4640-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/index.html

Good enough for now, or more tweaks?

**Comment by choldgraf** [comment]
looks a lot better IMO. The only thing that bugs me now is the buttons on the right being right-justified, and the funding text below being left-justified...do the buttons look weird if they're left-justified?

**Comment by Eric89GXL** [comment]
> do the buttons look weird if they're left-justified?

Yeah. You can test these things locally without needing to re-build the site. Even directly in chrome you can right-click -> "inspect" and then change CSS properties.

**Comment by choldgraf** [comment]
what about:

**Comment by Eric89GXL** [comment]
I don't like it as much personally

**Comment by Eric89GXL** [comment]
I tried it again, and I guess the left-align is *okay*, so if you really dislike the right I can change it

**Comment by choldgraf** [comment]
Maybe the thing that's bugging me is the checkbox - I think this is awkward w/ a right-justification, and also it makes me want to click it bc it looks like something interactive. What about just:

image

**Comment by Eric89GXL** [comment]
I like having the icons not just for eye-candy, but also as a visual anchor / bullet-type behavior.

**Comment by Eric89GXL** [comment]
If you want to iterate on this, we should do it on Gitter to avoid flooding GitHub notifications.

**Comment by choldgraf** [comment]
(in gitter now)

**Comment by agramfort** [comment]
@Eric89GXL did you try to put the carousel on the right of the MNE logo? it would allow to have the content boxes visible without scrolling down.

**Comment by dengemann** [comment]
Looks really nice, awesome work Chris & Eric! Almost uncannily professional
for academic software :) Can we put back some piece of demo code below to
expose a bit our API abstractions?
I agree we need to select the examples we want to advertise. The source
permutation clustering stats for example might not be among them.

**Comment by Eric89GXL** [comment]
We initially had it to the right but things were crowded.

I don't like having the code because it clutters the page. Ideally things
should fit on a single page without scrolling. The more things we add, the
harder it is to get where people want. People agreed on the initial mockup
that didn't have it, so I assume we already otherwise have consensus to get
rid of it.

Maybe instead I can make an advertisement by making the first carousel page
say you can go from raw to dSPM in 20 lines?

As per the previous discussion, can we can refine examples in a dedicated
follow-up PR? This one is focused on layout as much as possible.

**Comment by dengemann** [comment]
Yes. No reason to pur everything in 1 PR.

**Comment by agramfort** [comment]
On Fri, May 19, 2017 at 1:16 PM, Eric Larson <notifications@github.com>
wrote:

> We initially had it to the right but things were crowded.


can I see screenshot? sorry to be late again on the discussion ... :(

**Comment by Eric89GXL** [comment]
Had to re-do it. But it's something like this:

screenshot from 2017-05-19 09-58-47

This is what is currently proposed (same window size):

screenshot from 2017-05-19 10-00-12

I don't like the gallery-right layout as much, it doesn't flow as well to my eye. If the concern is sizing, I could make some of the current elements smaller.

**Comment by agramfort** [comment]
what if you make the 8 darks boxes bigger centered mimicking the
scikit-learn layout?

http://scikit-learn.org/stable/

**Comment by Eric89GXL** [comment]
Not sure if you're talking about the current layout or the left-right one. I'm also not sure what exactly you mean by mimicking their layout (break out the buttons into info boxes? what do you mean by centered, do you mean three columns?).

But here are both with bigger buttons at least:

screenshot from 2017-05-19 10-39-32

Current:

screenshot from 2017-05-19 10-44-12

**Comment by choldgraf** [comment]
Or do you mean something more like this @agramfort ?

http://predictablynoisy.com/mne-python/

This was one of the earlier versions of the PR but we had modified it to the current version because we didn't want it to be too cluttered.

File: doc/_static/style.css [hunk 1]
```diff
@@ -167,8 +198,90 @@ dt:target code {
...
+.carousel-caption {
+    text-shadow: 0 1px 1px rgba(0,0,0,1.0);
+    background-color: rgba(0,0,0,0.5);
+    box-shadow: 0 0 5px rgba(0,0,0,0.5);
+}
+.carousel-caption p {
+    font-size: 16px;
+}
+.tagline {
+  font-size: 24px !important;
```
Comment (agramfort) [inline doc/_static/style.css]: can you use:

.tagline {
  font-size: 22px !important;
  text-align: center;
}

**Comment by agramfort** [comment]
also we need to extend the bootstrap navbar breakdown width:

https://stackoverflow.com/questions/19827605/change-bootstrap-navbar-collapse-breakpoint-without-using-less

also can you make the fonts in the buttons bigger?

also when making the window less wide the interesting buttons appear at the very end of the page. I think they should appear maybe just after the logo so no need to scroll down on a phone or tablet.

sorry for the late feedback...

thanks a ton @Eric89GXL and @choldgraf for taking a stab at this. This is very very useful.

**Comment by Eric89GXL** [comment]
> .tagline {
> font-size: 22px !important;
> text-align: center;
> }

I like the size better, but this is too much centering IMO.

I'll look into the other fixes, too, and send an image soon I hope.

**Comment by Eric89GXL** [comment]
> also when making the window less wide the interesting buttons appear at the very end of the page. I think they should appear maybe just after the logo so no need to scroll down on a phone or tablet.

I tried this and it doesn't look good. I think that such a narrow case is going to be far less common than a fuller-width case. I'd rather not sacrifice the vast-majority case to make the rare case look better.

Instead I think I found a different way to display things that still stacks decently on smaller widths:

https://4661-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/index.html

**Comment by choldgraf** [comment]
Can we have a short skype on this sometime this week? Designing by committee is already hard enough, and doing so asynchronously is even more difficult.

my 2 cents on the buttons is that they should go under the image slider, since people generally respond better to pictures it should be given priority in the screen real-estate.

Also @Eric89GXL there's a type in the "receptive field" description for the slider. Just FYI.

**Comment by agramfort** [comment]
beautiful !

+1 for merge.

@choldgraf I give you the privilege to merge this !

**Comment by agramfort** [comment]
:beers: !

**Comment by teonbrooks** [comment]
looks great! nice work!! 🍻

**Pull request merged**